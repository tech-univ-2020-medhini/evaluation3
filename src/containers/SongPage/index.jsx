import React from 'react';
import PropTypes from 'prop-types';
import SongCard from '../../components/SongCard';

const SongPage = (props) => {
  const { songsOfGenre, genreName } = props;
  let songs;
  if (songsOfGenre !== undefined) {
    songs = songsOfGenre.map((song) => (
      // eslint-disable-next-line max-len
      <SongCard id={song.id} title={song.name} artists={song.artists} imageLink={song.albumArtUrl} key={song.id} />
    ));
  }


  return (
    <div className="Song-page">
      <p>
        {genreName}
      </p>
      <div className="Songs-container">
        {songs}
      </div>
    </div>
  );
};
SongPage.propTypes = {
  genreName: PropTypes.string.isRequired,
  songsOfGenre: PropTypes.arrayOf(PropTypes.object).isRequired,
};
export default SongPage;
