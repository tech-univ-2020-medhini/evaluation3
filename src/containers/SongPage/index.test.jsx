import React from 'react';
import { render } from '@testing-library/react';
import SongPage from './index';


describe('the songpage container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<SongPage />);

    expect(asFragment()).toMatchSnapshot();
  });
});
