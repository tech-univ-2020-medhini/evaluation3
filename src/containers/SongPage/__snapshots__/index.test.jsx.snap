// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the songpage container should render correctly 1`] = `
<DocumentFragment>
  <div
    class="Song-page"
  >
    <p />
    <div
      class="songs-container"
    />
  </div>
</DocumentFragment>
`;
